### DISCUSSION TITLE
Detect if TSML_UI has loaded ?

### DISCUSSION THREAD

**Discussion by k8fish3r** [code4recovery/12-step-meeting-list#1499] [Q&A] [answered]
Hi,

I've received feedback some browsers(potentially linked to the machines configuration) don't see TSML - rendering.

There is a "widget" block above which renders, but the end users machine (mac, running chrome) doesn't see TSML UI at all.

Rare I imagine, I was wondering is there anyway to detect:
- TSML_UI meeting list
- TSML_UI individual meeting page

has / hasn't completed ? 

Its not my machine / and I cant replicate - thinking if I can detect when rendering has finished on both page types - (or not) I can nudge the user to try another browser / device. 
The information I have is they use a macOS 10.11.6, and Google Chrome: OS X 10.11

Any thoughts welcome, for the meeting page, I did try to detect when animation to switch between list view had stopped/started (via animationend) . But I wasn't 100% sure which DOM objects to monitor. 

Web page: https://codauk.org/meetings/?tsml-day=any

Does anyone *not* see the meetings list render? Come across this ? Thinking its peculiar - dont believe Ive seen this arise on the discussion forum - despite the huge number of sites running TSML.

**Comment by joshreisner** [comment]
hi, that's right, that's not a report we hear very often! sounds like there was a javascript error that prevented the list from rendering. 

i'm not aware of a way to collect data on how much that's happening, but the DOM object to monitor would be the `div#tsml-ui`. maybe other users on here know of services to detect and record javascript errors?

it's great that you were able to get the user's OS information, looks like that was released 9 years ago (September 30, 2015). 

some things to try:
* ask them if other browsers, eg safari, work
* ask them if they can see the meeting list at aasanjose.org/meetings
* ask them if they can [check their javascript console](https://www.coursera.support/s/article/learner-000001653-How-to-open-the-Javascript-console?language=en_US) 

i do see this one error that isn't blocking me from seeing the meetings page, looks like it's related to the "javascript infinity module" - not sure what that is but might be worth seeing if that can be turned off?

<img width="822" alt="Screenshot 2024-08-25 at 10 16 06 AM" src="https://github.com/user-attachments/assets/7b58674b-3680-4312-974b-df399b513614">

**Comment by k8fish3r** [comment]
Apologies for taking so long to get back to you whilst in Google Analytics I remembered Id forgotten to answer and was finding some weird stats about visitors (about 70% mobile) - but REAL old - guess a portion of visitors are from older generations or not interested in constantly upgrade phone wars.

iOS visitors - weve about 3.5K active visitors on iOS 10. Guessing that could be at the root of the anomaly being witnessed as that OS is nearly 10 years old.

**Reply by gkovats** [reply, marked as answer]
We could dispatch a custom DOM event on `div#tsml-ui` on full load, so listeners on the page can track it, but that's really only half an answer. Otherwise, if you're good with chalking this up to a dated device, we'll close this issue. No doubt IE11 and old Opera versions exist in the wild with the same issues.

**Reply by k8fish3r** [reply]
Appreciate the teams time. Thank you.
